CONTRIBUTING.md

```diff
@@ -12,9 +12,81 @@ Make sure the tests pass:
 
 Make your change. Add tests for your change. Make the tests pass again.
 
+You can also install this project locally in editable mode to make changes and run any manual tests. This can be done by installing using the following pip command:
+
+```
+pip install -e <path_to_cloned_project_dir>
+```
+
 Push to your fork and [submit a pull request][pr].
 
 [pr]: https://github.com/your-username/datadogpy/compare/DataDog:master...master
 
 At this point you're waiting on us. We may suggest some changes or
 improvements or alternatives.
+
+# Adding new API endpoints
+This section outlines the process for adding a new endpoint to this API client.
+
+Lets use the example of creating an endpoint for `Hosts`. For this example endpoint, we will accept a GET and DELETE request at the `/hosts` endpoint as well as a GET resquest at the `hosts/totals` endpoint. **NOTE:** This endpoint is just an exmaple and doesn't describe the existing `hosts` resource.
+
+First start by adding a new file to the `datadog -> api` folder for our new endpoint `hosts.py`. We'll use the following simple class structure:
+
+```
+from datadog.api.resources import (
+    GetableAPIResource,
+    DeletableAPIResource
+)
+
+class Hosts(GetableAPIResource, DeletableAPIResource):
+    """
+    A wrapper around Hosts HTTP API.
+    """
+    _resource_name = 'hosts'
+```
+
+Each class has the above simple structure, most importantly the following two pieces:
+
+* A `_resource_name` - Indicates the uri of the api.
+* A set of classes to inherit from. This is where the get/post/put/delete request code is defined for you. Available options are:
+  * CreateableAPIResource - Wrapper class for providing a `POST` request for your class, implementing a `create` method.
```

Comment:
Let's add link to corresponding classes

Reply:
I think you could create an Array for all available options. Since it separates the options from the class requirement from a visual level.

Reply:
Turned it into a table.